distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/APMJvmOptions.java

```diff
@@ -187,20 +187,12 @@ static String agentCommandLineOption(Path agentJar, Path tmpPropertiesFile) {
     static void extractSecureSettings(SecureSettings secrets, Map<String, String> propertiesMap) {
         final Set<String> settingNames = secrets.getSettingNames();
         for (String key : List.of("api_key", "secret_token")) {
-            for (String prefix : List.of("telemetry.", "tracing.apm.")) {
-                if (settingNames.contains(prefix + key)) {
-                    if (propertiesMap.containsKey(key)) {
-                        throw new IllegalStateException(
-                            Strings.format("Duplicate telemetry setting: [telemetry.%s] and [tracing.apm.%s]", key, key)
-                        );
-                    }
-
-                    try (SecureString token = secrets.getString(prefix + key)) {
-                        propertiesMap.put(key, token.toString());
-                    }
+            String prefix = "telemetry.";
```

Comment:
Nit: There's not much reason to separate out the prefix anymore. It could just be prepended to each of the keys in the list above.

Or if we want to keep the prefix, perhaps we could reference [this one](https://github.com/elastic/elasticsearch/pull/119926/files#diff-8e5ca1315006cc00c58ef42d4bfc14f50b30d11567a845f8e551ecf3e52c830bR100)?

Reply:
`propertiesMap` contains those keys without prefix, so still easier to keep the prefix separate

Reply:
If depending on the prefix you've linked, we'd need a dependency of server-cli on the APM module.
That doesn't seem worth it.